homework/index.js

```diff
@@ -30,17 +32,152 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
-      }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+  function renderError(error) {
+    const wrapper = document.getElementById('wrapper');
+    wrapper.innerHTML = '';
+    createAndAppend('div', wrapper, {
+      text: error.message,
+      class: 'alert',
+    });
+  }
+
+  function addRepoInfoRow(parent, label, content) {
+    const row = createAndAppend('tr', parent);
+    createAndAppend('th', row, {
+      text: label,
+      class: 'table-header',
     });
+    createAndAppend('td', row, {
+      text: content,
+    });
+    return row;
+  }
+
+  function renderRepoInfo(repository) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const repoTable = createAndAppend('table', leftSide);
+    const repoTableBody = createAndAppend('tbody', repoTable);
+    const repoNameRow = addRepoInfoRow(repoTableBody, 'Repository', '');
+    createAndAppend('a', repoNameRow.lastChild, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name,
+    });
+    addRepoInfoRow(repoTableBody, 'Description', repository.description);
+    addRepoInfoRow(
+      repoTableBody,
+      'Created At',
+      new Date(repository.created_at).toLocaleString('en-GB'),
+    );
+    addRepoInfoRow(
+      repoTableBody,
+      'Updated At',
+      new Date(repository.updated_at).toLocaleString('en-GB'),
+    );
+    addRepoInfoRow(repoTableBody, 'Forks', repository.forks_count);
+    addRepoInfoRow(repoTableBody, 'Watchers', repository.watchers_count);
   }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+  function fetchAndRender(repository) {
+    const rightSide = document.getElementById('right-side');
+    rightSide.innerHTML = '';
+    fetchJSON(repository.contributors_url)
+      .then(contributors => {
+        const contributorsTitle = createAndAppend('h3', rightSide, {
+          text: 'Contributors',
+          class: 'contributors-title',
+        });
+        if (!contributors.length) {
+          contributorsTitle.textContent = 'No Contributor So Far';
+          return;
+        }
+        const contributorsList = createAndAppend('ul', rightSide, {
+          class: 'contributors-list',
+        });
+        contributors.forEach(contributor => {
+          const listItem = createAndAppend('li', contributorsList, {
+            class: 'contributor',
+            tabindex: 0,
```

Comment:
👍 To include the `li` in the tab order so it can be navigated to with the keyboard.